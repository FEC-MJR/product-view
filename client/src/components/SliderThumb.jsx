import React from 'react'
import styled from 'styled-components'
import CreateImageCell from './CreateImageCell.jsx'

const StyledImageThumbnail = styled.div`
  background: #e8e7e6;
`;

//Creates thumbnail images for carousel - props are passed down from <Carousel/>
var SliderThumb = (props) => {
  //map through props images and set key/image for CreateImageCell component. ImageThumbnail will render below with returned images.
  const ImageThumbnail = props.product.images.map((imageSource, index) => (
    <CreateImageCell key={index} index={index} imageSource={imageSource} activeCell={props.product.activeSlide}/>
  ))

  return(
    <StyledImageThumbnail className='slider-thumb'>{ImageThumbnail}</StyledImageThumbnail>
  )
}

export default SliderThumb