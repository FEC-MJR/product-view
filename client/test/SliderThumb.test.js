import React from 'react'
import { shallow, mount, render } from 'enzyme'
import SliderThumb from '../src/components/SliderThumb.jsx'

describe("Component testing for <SliderThumb/>", () => {

  const props = {
    product:  ["https://hrphxfec.s3-us-west-1.amazonaws.com/Kanken+Laptop+13/kanken_acorn.jpg"]
  }

  test("<SliderThumb/> should render", () => {

    const wrapper = shallow(<SliderThumb {...props}/>)
    expect(wrapper)
  })
})