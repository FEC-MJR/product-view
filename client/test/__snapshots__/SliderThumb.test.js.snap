// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing for <SliderThumb/> <SliderThumb/> snapshot 1`] = `
<div
  className="slider-thumb"
>
  <img
    className="sc-AxjAm jLqTMr"
    src="https://hrphxfec.s3-us-west-1.amazonaws.com/Kanken+Laptop+13/kanken_acorn.jpg"
  />
</div>
`;
